doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+        container_local_output_path='/local/path/for/data/emission'
+        hook_parameters={
+            "save_interval": "10"
+        },
+        collection_configs=[
+            collection_config_1, collection_config_2
+        ]
+    )
+
+In the above sample code, the ``save_interval`` of ``10`` will be applied for storing both collections.
+
+Note that the value set in the ``collection_parameters`` for a parameter will override the corresponding value of the same parameter in the ``hook_parameters``. For example, in the above sample code, if ``collection_config_2`` had the value of ``save_interval`` set something other than ``10``, say, ``20``, then the tensors for that collection would have been saved with step interval ``20`` while those for ``collection_config_1`` would still be saved with ``10``.
+
+Continuous analyses through Rules
+=================================
+
+In addition to collecting the debugging data, Amazon SageMaker Debugger provides the capability for you to analyze it in a streaming fashion through "rules". SageMaker Debugger 'Rules' are pieces of code which encapsulate the logic for analyzing the debugging data.
+
+SageMaker Debugger provides a set of built-in rules curated by data scientists and engineers at Amazon. There is additional support for bringing in custom rule source codes for evaluation. In the following sections, you'll learn how to use either while training your model.
+
+Relationship between debugger hook and rules
+--------------------------------------------
+
+Using SageMaker Debugger is, broadly, a two-pronged approach. One one hand you have production of debugging data, which is done through the Debugger Hook and on the other hand you have the consumers which can be rules (for continuous analyses) or SageMaker Debugger SDK (for interactive analyses).
+
+Usually, production of data is independent of the manner in which it's consumed (and vice-versa.) For example, you can configure the debugging hook to store only the collection "gradients" and configure the rules to operate on some other collection, say, "weights". While this is possible, it's not ideal since you won't get any meaningful insight into the training process through this.
+
+For efficient debugging, you'd want to configure your debugging hook to produce and store data you care about and have the rules operate on that particular data and make sense of it. This way, you ensure that the Debugger is utilized to its maximum potential in detecting anomalies. In this sense, there is a loose binding between the hook and the rules.
+
+Normally, you'd achieve this binding for a training job by providing values for both ``debugger_hook_config`` and ``rules`` in your estimator. However, SageMaker Debugger simplifies this by allowing you to specify the collection configuration within the ``Rule`` object itself. This way, you don't have to specify ``debugger_hook_config`` in your estimator separately.
```

Comment:
Suggested edit: "For efficient debugging, configure your debugging hook to produce and store the data that you care about and employ rules that operate on that particular dataset."

Reply:
* [@Vikas-kum] Not putting an example here since we're discouraging this approach. Putting an example will only confuse the users.
* [@aws-rbs] Incorporated the suggested edit with slight modification to the word data -> "debugging data" and "dataset" -> "data" ("dataset" is colloquially associated with model training dataset)

Reply:
RE (it's a good practice to keep the debug artifacts separate for different training jobs for correct analyses later on.)" 
This point seems too important to be parenthetical. How about:
"To enable correct analysis of different training jobs, it is a best and essential practice to store the debug artifacts for these jobs separately."